### ISSUE TITLE
Vagrant installed via libvirt script is out of date

### ISSUE THREAD

**Issue by p-rintz** [ipspace/netlab#1022]
Hello,

the libvirt script installs version 2.3.4 of Vagrant in this line:
https://github.com/ipspace/netlab/blob/a3c46c692cb05f772bf17ee87af0cf787bdcf796/netsim/install/libvirt.sh#L67

This leads to an issue when installing vagrant-libvirt due to an out of date ruby version that is packaged with that version of Vagrant.

```
nokogiri requires Ruby version >= 3.0, < 3.4.dev. The current ruby version is 2.7.6.219.
```

Thanks for the great bgp labs website!

**Comment by ipspace** [comment]
The Vagrant dependency hell strikes again :( We had to pin the vagrant-libvirt version in https://github.com/ipspace/netlab/commit/c22ab4a30cdfcc3c0ae81ffcd8b8775142c7aee6 to fix another dependency SNAFU and then pin the vagrant version because they messed up big time (https://github.com/ipspace/netlab/commit/2409966f8f53ede6fa6de8ad40aa918988f48268).

Anyway, looks like it's time to pin a different version. Did you manage to get a working installation? If so, what version of Ubuntu, Vagrant and vagrant-libvirt works for you?

Thank you! Ivan

**Comment by ipspace** [comment]
OTOH, I just figured out that I've been running Vagrant 2.4.0 and vagrant-libvirt 0.12.2 on my production machine for quite a while 🤦‍♂️ so I guess it's safe to move to those versions 🤷‍♂️

Will run some tests tomorrow and then push out 1.7.1-post1 -- your feedback on what works for you is still highly appreciated ;)

**Comment by p-rintz** [comment]
Same as you.
0.12.2 and 2.4.0

I simply removed the pinned versions in the script and it ran through (for now).

**Comment by ipspace** [comment]
> Same as you. 0.12.2 and 2.4.0
> 
> I simply removed the pinned versions in the script and it ran through (for now).

That's been fixed (I changed the pinned versions, still don't trust Vagrant et co won't break). BTW, were you using Ubuntu 20.04 or 22.04? 22.04 seems to use Ruby 3 whereas 20.04 is still on Ruby 2.7.

BTW, the breakage was caused by a new version of nokogiri (release on December 16th 2023) that requires Ruby 3.0. No idea why the whole thing works on Ubuntu 20.04 though, maybe Vagrant brings its own version of Ruby?
